include/seqan3/alphabet/mask/masked.hpp

```diff
@@ -55,36 +55,52 @@ namespace seqan3
  * masked alphabet. It allows one to specify which portions of a sequence should be masked,
  * without losing additional information by replacing the sequence directly.
  */
-
  template <typename sequence_alphabet_t, typename mask_t = mask>
+//!\cond
     requires alphabet_concept<sequence_alphabet_t>
- struct masked :
+//!\endcond
+class masked :
     public cartesian_composition<masked<sequence_alphabet_t, mask_t>,
-        sequence_alphabet_t, mask_t>
+                                 sequence_alphabet_t, mask_t>
 {
+private:
+    //!\brief The base type.
+    using base_type = cartesian_composition<masked<sequence_alphabet_t, mask_t>,
+                                            sequence_alphabet_t, mask_t>;
+public:
     //!\brief First template parameter as member type.
     using sequence_alphabet_type = sequence_alphabet_t;
 
     //!\brief Equals the char_type of sequence_alphabet_type.
     using char_type = underlying_char_t<sequence_alphabet_type>;
 
-    /*!\name Write functions
+    /*!\name Constructors, destructor and assignment
      * \{
      */
-    //!\brief Directly assign the sequence letter.
-    constexpr masked & operator=(sequence_alphabet_type const l) noexcept
-    {
-        get<0>(*this) = l;
-        return *this;
-    }
+    masked() = default;
+    constexpr masked(masked const &) = default;
+    constexpr masked(masked &&) = default;
+    constexpr masked & operator =(masked const &) = default;
+    constexpr masked & operator =(masked &&) = default;
+    ~masked() = default;
 
-    //!\brief Directly assign the mask letter.
-    constexpr masked & operator=(mask const l) noexcept
-    {
-        get<1>(*this) = l;
-        return *this;
-    }
+    using base_type::base_type; // Inherit non-default constructors
+
+    using base_type::operator=; // Inherit non-default assignment operators
+
+    //!\copydoc cartesian_composition::cartesian_composition(alphabet_type const alph)
+    SEQAN3_DOXYGEN_ONLY(( constexpr masked(alphabet_type const alph) {} ))
+    //!\copydoc cartesian_composition::cartesian_composition(alphabet_subtype const alph)
+    SEQAN3_DOXYGEN_ONLY(( constexpr masked(alphabet_subtype const alph) {} ))
+    //!\copydoc cartesian_composition::operator=(alphabet_type const alph)
+    SEQAN3_DOXYGEN_ONLY(( constexpr masked & operator=(alphabet_type const alph) {} ))
+    //!\copydoc cartesian_composition::operator=(alphabet_subtype const alph)
+    SEQAN3_DOXYGEN_ONLY(( constexpr masked & operator=(alphabet_subtype const alph) {} ))
```

Comment:
Strictly speaking these should be templates, too, but if the generated documentation looks ok, you can keep it as-is.

Reply:
![grafik](https://user-images.githubusercontent.com/7955878/44464570-6db97a80-a61b-11e8-8bd2-230e17f62e7e.png)


Fine enough?